### PR TITLE
reduce number of calls to update shard

### PR DESCRIPTION
**Pull request by wxing1292**
solve #634

### CODE REVIEW

File: service/history/service.go [hunk 1]
```diff
@@ -107,30 +110,31 @@ func NewConfig(dc *dynamicconfig.Collection, numberOfShards int) *Config {
 		TimerTaskMaxRetryCount:                             5,
 		TimerProcessorGetFailureRetryCount:                 5,
 		TimerProcessorCompleteTimerFailureRetryCount:       10,
-		TimerProcessorUpdateAckInterval:                    10 * time.Second,
-		TimerProcessorForceUpdateInterval:                  10 * time.Minute,
+		TimerProcessorUpdateShardTaskCount:                 100,
```
Comment (samarabbas) [inline service/history/service.go]: Let's file a task to make these dynamic config knobs.

File: service/history/service.go [hunk 2]
```diff
@@ -107,30 +110,31 @@ func NewConfig(dc *dynamicconfig.Collection, numberOfShards int) *Config {
...
-		TransferProcessorForceUpdateInterval:               10 * time.Minute,
-		TransferProcessorCompleteTransferInterval:          1 * time.Second,
-		TransferProcessorStandbyTaskDelay:                  0 * time.Minute,
 		TransferTaskWorkerCount:                            10,
 		TransferTaskMaxRetryCount:                          100,
 		TransferProcessorCompleteTransferFailureRetryCount: 10,
+		TransferProcessorUpdateShardTaskCount:              100,
+		TransferProcessorMaxPollInterval:                   60 * time.Second,
+		TransferProcessorUpdateAckInterval:                 1 * time.Minute,
+		TransferProcessorCompleteTransferInterval:          1 * time.Second,
```
Comment (samarabbas) [inline service/history/service.go]: 1 second for complete sounds pretty low, considering we have 16K shards.  I think it should also be notification based when there is something to cleanup.  
Let's file a task to review all service configs.

File: service/history/service.go (hunk 1 above)
Comment (wxing1292) [reply service/history/service.go]: see #592

File: service/history/service.go (hunk 2 above)
Comment (wxing1292) [reply service/history/service.go]: #741

**Pull request merged**